### PR TITLE
chore: run e2e tests in CI

### PR DESCRIPTION
**Pull request by bryanchriswhite**
### Changes

- use pre-built fetchd base-image (faster)
- use CI-specific docker compose file
- ensure all node_modules directories are covered by dockerignore file
- add workflow to exercise end-to-end tests

### Dependencies

- [x] #74

### CODE REVIEW

**Comment by bryanchriswhite** [comment]
I think the end-to-end tests are failing because both `TestContractSwap` and `TestContractExecution` inherit from `BaseContract` which deploys the bridge contract in `setUpClass`. If we use cosmpy to check if a contract with that address already exists, we should be able to use the same address when running the tests all together or in isolation.

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+        run: sudo apt update
+      - name: Install OS dependencies
+        run: sudo apt install -y python3-dev
+
+      - uses: actions/checkout@v3
+        with:
+          submodules: recursive
+
+      - name: Start-up local test environment
+        uses: isbang/compose-action@v1.2.0
```
Comment (daeMOn63) [inline .github/workflows/test.yml:26]: From https://github.com/isbang/compose-action#compose-file, we probably need to specify the right compose file to use?

```suggestion
        uses: isbang/compose-action@v1.2.0
        with:
          compose-file: "./ci-compose.yml"
```

Comment (daeMOn63) [reply .github/workflows/test.yml:26]: (this might also fix the error from the failed run)

Comment (bryanchriswhite) [reply .github/workflows/test.yml:26]: Thanks @daeMOn63, good catch!

**Review by daeMOn63 — APPROVED** [review]

**Pull request merged**